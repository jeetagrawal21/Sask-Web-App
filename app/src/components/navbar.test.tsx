// TEST FILE FOR NAVBAR COMPONENT 
// HAS CASES FOR SNAPSHOT TESTING

import React from 'react';
import { render } from '@testing-library/react';
import Navbar from './navbar';

/**
 * Descrpition: Test function for Navbar component.
 * 
 * Pre-Condition: None
 * 
 * Post-Condition: None
 * 
 * Return: None
 */

describe('Navbar component', () => {

    // Test Case 1 : Checks if the navbar component renders correctly
    it('should render properly', () => {
      const { container } = render(<Navbar />);
      expect(container).toMatchSnapshot();
    });
  
    // Test Case 1 : Checks if all links renders correctly
    it('should render all links properly', () => {
      const { getByText } = render(<Navbar />);
      expect(getByText('About')).toBeInTheDocument();
      expect(getByText('Contact us')).toBeInTheDocument();
      expect(getByText('Help')).toBeInTheDocument();
    });
  });


  /**
    test('snapshot test for About link', () => {
    const component = create(<NavBar />);
    const aboutLink = component.root.findByProps({ href: '/about' });
    expect(aboutLink.props.children).toBe('About');
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('snapshot test for Contact Us link', () => {
    const component = create(<NavBar />);
    const contactLink = component.root.findByProps({ href: '/contactus' });
    expect(contactLink.props.children).toBe('Contact Us');
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('snapshot test for Help link', () => {
    const component = create(<NavBar />);
    const contactLink = component.root.findByProps({ href: '/help' });
    expect(contactLink.props.children).toBe('Help');
    expect(component.toJSON()).toMatchSnapshot();
   */