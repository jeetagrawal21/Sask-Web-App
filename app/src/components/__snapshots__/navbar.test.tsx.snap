// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render properly 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      class="nav-link"
      href="#"
    >
      About
    </a>
    <a
      class="nav-link"
      href="#"
    >
      Contact us
    </a>
    <a
      class="nav-link"
      href="#"
    >
      Help
    </a>
  </nav>
</div>
`;

exports[`Navbar component should render properly 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      class="nav-link"
      href="#"
    >
      About
    </a>
    <a
      class="nav-link"
      href="#"
    >
      Contact us
    </a>
    <a
      class="nav-link"
      href="#"
    >
      Help
    </a>
  </nav>
</div>
`;
